target_chains/ethereum/contracts/contracts/entropy/Entropy.sol

```diff
@@ -301,6 +320,88 @@ abstract contract Entropy is IEntropy, EntropyState {
         }
     }
 
+    function revealAndCall(
+        address provider,
+        uint64 sequenceNumber,
+        bytes32 userRandomness,
+        bytes32 providerRevelation
+    ) public returns (bytes32 randomNumber) {
+        EntropyStructs.Request storage req = findRequest(
```

Comment:
why not calling the reveal function internally, same as you did with the request function?